OsmAnd/src/net/osmand/plus/helpers/WaypointHelper.java

```diff
@@ -357,8 +357,9 @@ public AlarmInfo calculateMostImportantAlarm(RouteDataObject ro, Location loc, M
 					if (info != null) {
 						if (info.getType() != AlarmInfoType.SPEED_CAMERA || showCameras) {
 							long ms = System.currentTimeMillis();
-							if (ms - announcedAlarmTime > 50 * 1000) {
-								announcedAlarmTime = ms;
+							Long timeLastAlarm = lastAnnouncedAlarmsTime.get(info.getType());
```

Comment:
From OpenGL thread

Reply:
Essentially this block do the same as in announceVisibleLocations but from another thread. These blocks supress each other.